### PR TITLE
[PPS] Removed all 'get' prefixes from PPS DataFormat objects

### PR DESCRIPTION
**Pull request by forthommel**
#### PR description:

This purely technical/aesthetic PR addresses the proposal in #25690 to drop all `get` prefixes in PPS data formats accessors.

#### PR validation:

Compiles fine, backward-compatible with previously produced samples.

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- [x] verify that the PR is really intended for the chosen branch
- [x] verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- [x] verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

cc: @jan-kaspar @fabferro

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by christopheralanwest** [comment]
please test

The failure of 1000.0 step3 seems to be a transient file access issue

File: RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc [hunk 1]
```diff
@@ -130,8 +130,8 @@ double ProtonReconstructionAlgorithm::ChiSquareCalculator::operator()(const doub
     const double y = k_out.y - oit->second.y0;
 
     // calculate chi^2 contributions, convert track data mm --> cm
-    const double x_diff_norm = (x - track->getX() * 1E-1) / (track->getXUnc() * 1E-1);
-    const double y_diff_norm = (y - track->getY() * 1E-1) / (track->getYUnc() * 1E-1);
+    const double x_diff_norm = (x - track->x() * 1E-1) / (track->yUnc() * 1E-1);
```
Comment (perrotta) [inline RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: ```xUnc()```

File: RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc [hunk 2]
```diff
@@ -130,8 +130,8 @@ double ProtonReconstructionAlgorithm::ChiSquareCalculator::operator()(const doub
...
+    const double y_diff_norm = (y - track->x() * 1E-1) / (track->yUnc() * 1E-1);
```
Comment (perrotta) [inline RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: ```y()```

**Review by perrotta — COMMENTED** [review]
These are the only two typos I found during a visual check.
Once fixed, to be sure that nothing else has escaped one should also verify the output of some meaningful comparison.

File: RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc (hunk 1 above)
Comment (forthommel) [reply RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: Thank you very much @perrotta! Fixed in 
9c02087

File: RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc (hunk 2 above)
Comment (forthommel) [reply RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: Thank you very much @perrotta !
Fixed in 9c02087

**Comment by jfernan2** [comment]
please test

**Comment by perrotta** [comment]
Same error in wf 136.8311 as in the previous set of tests, most likely correlated with this PR

```
...
#6  0x00002b7f1dfc4c81 in TClass::GetDataMemberOffset(char const*) const () from /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_0_X_2019-10-24-1100/external/slc7_amd64_gcc820/lib/libCore.so
#7  0x00002b7f4948c066 in ROOT::read_CTPPSPixelLocalTrack_0(char*, TVirtualObject*) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-10-24-1100/lib/slc7_amd64_gcc820/libDataFormatsCTPPSReco.so
...
```

**Comment by forthommel** [comment]
> Same error in wf 136.8311 as in the previous set of tests, most likely correlated with this PR
> 
> ```
> ...
> #6  0x00002b7f1dfc4c81 in TClass::GetDataMemberOffset(char const*) const () from /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_0_X_2019-10-24-1100/external/slc7_amd64_gcc820/lib/libCore.so
> #7  0x00002b7f4948c066 in ROOT::read_CTPPSPixelLocalTrack_0(char*, TVirtualObject*) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-10-24-1100/lib/slc7_amd64_gcc820/libDataFormatsCTPPSReco.so
> ...
> ```

You are perfectly right! Digging in this direction allowed me to discover I was messing with one attribute version converter too much for this object... Should be fixed in 17eaf38 (local test running in parallel).

**Comment by christopheralanwest** [comment]
please test

**Comment by forthommel** [comment]
It seems that a slight adaptation of the validation script is indeed required to propagate this removal of our infamous `get` prefix. Introduced in https://github.com/cms-sw/cms-bot/pull/1217.

**Comment by perrotta** [comment]
> It seems that a slight adaptation of the validation script is indeed required to propagate this removal of our infamous `get` prefix. Introduced in [cms-sw/cms-bot#1217](https://github.com/cms-sw/cms-bot/pull/1217).

Thank you @forthommel , also for providing the needed update for the validation script.
Indeed all differences reported in the reco outputs by the automatic comparisons refer to the accessors which had "get" in their previous name.
The fact that DQM comparisons do not show any difference in any of the tested workflows can make us confident that there were no hidden mistakes in propagating the renamings in the code. I think that re-running the tests with the updated validation script should easily provide us with a few distibutions to compare to the previous reco outputs, and therefore further confirming that there were no other typos escaped to our eyes: let try it.

**Comment by perrotta** [comment]
please test with  cms-sw/cms-bot#1217

**Comment by forthommel** [comment]
Thank you, @perrotta ! I was curious to see whether the validation script logic would declare its test as a failure or a success, as to me the 1-to-1 comparison with the baseline is not assured to be fulfilled (can it be that the script "sees" the disappearance of all `getXXX()` histograms and the apparition of new `xxx()` histogram without any prior knowledge of the conversion rule?)

**Comment by slava77** [comment]
> I think that re-running the tests with the updated validation script should easily provide us with a few distibutions to compare to the previous reco outputs, and therefore further confirming that there were no other typos escaped to our eyes: let try it.

I doubt that it will work.
The PR was made in a way that now the old format plots will not be made.

**Comment by perrotta** [comment]
> > I think that re-running the tests with the updated validation script should easily provide us with a few distibutions to compare to the previous reco outputs, and therefore further confirming that there were no other typos escaped to our eyes: let try it.
> 
> I doubt that it will work.
> The PR was made in a way that now the old format plots will not be made.

I'd expect that we'll get empty/full histograms with reverted colors than the ones of the previous comparison, that we can compare "by eye" at that point. But maybe I'm just wrong: let see what sort out from this comparison.

**Comment by perrotta** [comment]
> > > I think that re-running the tests with the updated validation script should easily provide us with a few distibutions to compare to the previous reco outputs, and therefore further confirming that there were no other typos escaped to our eyes: let try it.
> > 
> > 
> > I doubt that it will work.
> > The PR was made in a way that now the old format plots will not be made.
> 
> I'd expect that we'll get empty/full histograms with reverted colors than the ones of the previous comparison, that we can compare "by eye" at that point. But maybe I'm just wrong: let see what sorts out from this comparison.

Ok, you were right @slava77 (as usal...). Do you think that the update to the monitoring script should be made differently?

@forthommel , about this PR: could you please check and report here about (lack of) differences between distributions with the "old" variables and the "new" ones? I think that it may be sufficient to compare with the baseline the outputs of one of the workflows run in the automatic jenkins tests

**Comment by cvuosalo** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by forthommel** [comment]
@perrotta, @slava77 
A long and tedious 1-to-1 comparison of most of the meaningful variables (i.e. all variables propagated all along several steps of the PPS reconstruction chain, and a few isolated variables) between reference ([here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-baseline-tests/CMSSW_11_0_X_2019-10-24-2300/slc7_amd64_gcc820/-GenuineIntel/matrix-results/)) and this modified release ([here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c7ea9b/3150/runTheMatrix-results/)) for workflows 136.85, 136.731, and 136.788 shows no difference.

**Comment by perrotta** [comment]
> @perrotta, @slava77
> A long and tedious 1-to-1 comparison of most of the meaningful variables (i.e. all variables propagated all along several steps of the PPS reconstruction chain, and a few isolated variables) between reference ([here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-baseline-tests/CMSSW_11_0_X_2019-10-24-2300/slc7_amd64_gcc820/-GenuineIntel/matrix-results/)) and this modified release ([here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c7ea9b/3150/runTheMatrix-results/)) for workflows 136.85, 136.731, and 136.788 shows no difference.

Thank you @forthommel

**Comment by perrotta** [comment]
+1
- All changes in code consist in the renaming of the getter methods, as well as a few member variables, as planned
- Jenkins tests pass and show no differences
_ No differences observed also in a "long and tedious 1-to-1 comparison of most of the meaningful variables" by Laurent, further certifies that there were no typing errors in the translation.

**Comment by ggovi** [comment]
+1

**Comment by fabiocos** [comment]
@christopheralanwest @tlampen @pohsun could you please check this PR? This should better enter in parallel with https://github.com/cms-sw/cms-bot/pull/1217 to get a clean comparison

**Comment by tlampen** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**